### PR TITLE
Add retry to iOS/tvOS/Android test on virtual device

### PR DESCRIPTION
**Pull request by sunmou99**
Retry (maximum 3 times) on error, e.g. integration test ended unexpectedly .
Won't retry if tests failed.

### CODE REVIEW

**Comment by DellaBitta** [comment]
I feel like this should post a comment to the PR, or log somewhere, that a flake occurred but it eventually passed.  So that we can keep a look out for flake.

**Comment by sunmou99** [comment]
> I feel like this should post a comment to the PR, or log somewhere, that a flake occurred but it eventually passed. So that we can keep a look out for flake.

Talked with Jon. For now, on the testapp level, we won't add retry if test failed. Jon will add retry in each flaky tests.
But I do love your idea. Because retry doesn't resolve the flakiness. Will create a ticket.

File: scripts/gha/test_simulator.py [hunk 1]
```diff
@@ -420,13 +418,18 @@ def _get_bundle_id(app_path, config):
...
   """Run gameloop test and collect test result."""
   logging.info("Running apple gameloop test: %s, %s, %s, %s", bundle_id, app_path, gameloop_app, device_id)
   _install_apple_app(app_path, device_id)
   _run_xctest(gameloop_app, device_id)
   logs = _get_apple_test_log(bundle_id, app_path, device_id)
   _uninstall_apple_app(bundle_id, device_id)
+  if retry > 1:
+    results = test_validation.validate_results_cpp(logs)
+    if not results.complete:
+      return _run_apple_gameloop_test(bundle_id, app_path, gameloop_app, device_id, retry=retry-1)
```
Comment (DellaBitta) [inline scripts/gha/test_simulator.py:431]: Is it worth appending something to the logs that notes the number of retries?

File: scripts/gha/test_simulator.py [hunk 2]
```diff
@@ -462,14 +465,13 @@ def _get_apple_test_log(bundle_id, app_path, device_id):
...
-  logging.info("Reading file: %s", path)
-  logging.info("File contant: %s", test_result)
-  return test_result
-
+  if os.path.isfile(path):
+    with open(path, "r") as f:
+      test_result = f.read()
 
+    logging.info("Reading file: %s", path)
+    logging.info("File contant: %s", test_result)
```
Comment (DellaBitta) [inline scripts/gha/test_simulator.py:473]: nit: `content`

**Review by DellaBitta — APPROVED** [review]

File: scripts/gha/test_simulator.py (hunk 1 above)
Comment (sunmou99) [reply scripts/gha/test_simulator.py:432]: added

**Review by DellaBitta — APPROVED** [review]

**Pull request merged**